cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2292,6 +2292,7 @@ macro(build_benchmark)
   file(MAKE_DIRECTORY "${GBENCHMARK_INCLUDE_DIR}")
 
   add_library(benchmark::benchmark STATIC IMPORTED)
+  target_compile_definitions(benchmark::benchmark INTERFACE BENCHMARK_STATIC_DEFINE)
```

Comment:
My previous attempt at this change mistakenly deleted a crucial line. :sweat:

Fingers crossed that this will fix the MSVC issue.